### DISCUSSION TITLE
TSDF integration results in very sparse mesh

### DISCUSSION THREAD

**Discussion by raabuchanan** [isl-org/Open3D#5840] [General]
Hello, I'm trying to use Open3d to integrate several uncolored pointclouds into a TSDF. The pointclouds are already aligned using ICP and expressed in the same world frame. My problem is that when integrating the TSDF I get a very sparse mesh, even sparser than the pointcloud. I was expecting the resulting mesh to connect the surface points.

Here is TSDF Mesh:
![Screenshot from 2023-01-10 16-31-31](https://user-images.githubusercontent.com/5701905/211608243-a4767210-7082-4bfd-b5e0-54dae0e50418.png)

Here is TSDF Pointcloud:
![Screenshot from 2023-01-10 16-31-54](https://user-images.githubusercontent.com/5701905/211608321-cbe54b39-9d3a-49d8-ac41-da0356ca556f.png)

Here is TSDF Voxel
![Screenshot from 2023-01-10 16-32-13](https://user-images.githubusercontent.com/5701905/211608395-6c093989-ec02-4336-89d3-98f33faefbf1.png)
l Pointcloud

My process has been as follows:
1. Project pointclouds to depth images using idealized camera intrinsic.
2. Convert depth images to RGBD images with gray color
3. Integrate RGBD images using ScalableTSDFVolume

Why am I getting such a sparse mesh from the TSDF? I've verified at every step the depth images and RGBD images look correct and can be converted back correctly to Pointclouds. I've included the code for TSDF integration below, thank you.

```
depths = [depth_img0, depth_img1]
camera_poses = read_trajectory("/home/russell/git/cat_and_maus/results/reconstructions/scan_3/vicon/segmented2/reconstructions/camera_poses.log")

volume = o3d.pipelines.integration.ScalableTSDFVolume(
    voxel_length=0.001,
    sdf_trunc=0.05,
    color_type=o3d.pipelines.integration.TSDFVolumeColorType.NoColor)

for i in range(len(camera_poses)):
    print("Integrate {:d}-th image into the volume.".format(i))

    depth = depths[i]

    depth_as_img =  o3d.geometry.Image(depth)
    
    rgbd = o3d.geometry.RGBDImage.create_from_color_and_depth(
        depth_as_img, depth_as_img, depth_scale=depth_scale, depth_trunc=depth_max, convert_rgb_to_intensity=True)

    print("Depth Image" + str(i) + " back as pointcloud (Test)")

    volume.integrate(
        rgbd,
        pinhole_intrinsic,
        np.linalg.inv(camera_poses[i].pose))


print("TSDF")
mesh = volume.extract_triangle_mesh()
mesh.compute_vertex_normals()
print(np.asarray(mesh.triangle_normals))
o3d.visualization.draw_geometries([mesh])
```

**Comment by AbbeGo** [comment]
Did you have solved this problem?  I currently meet the same problem.
